staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -171,16 +207,16 @@ func newTestHTTPServer(f AttachFunc, options *StreamOptions) *httptest.Server {
 	return server
 }
 
-func attach2Server(rawURL string, options StreamOptions) error {
+func attach2Server(ctx context.Context, rawURL string, options StreamOptions) error {
 	uri, _ := url.Parse(rawURL)
 	exec, err := NewSPDYExecutor(&rest.Config{Host: uri.Host}, "POST", uri)
 	if err != nil {
 		return err
 	}
 
-	e := make(chan error)
+	e := make(chan error, 1)
```

Comment:
Why do we need buffering here?

Reply:
Because the sender blocks if there's no buffer and receiver at the same time. `select` below would quit early if context is cancelled or timeout so that conditions are all met and stream goroutine leaks.

Reply:
Do you mean that without the buffering, if the `select` picks the `time.After` branch, then the `e <- exec.StreamWithContext(ctx, options)` goroutine will block and leak? Good catch.